MetadataExtractor/IO/ReaderInfo.cs

```diff
@@ -0,0 +1,694 @@
+﻿// Copyright (c) Drew Noakes and contributors. All Rights Reserved. Licensed under the Apache License, Version 2.0. See LICENSE in the project root for license information.
+
+using System;
+using System.IO;
+using System.Text;
+
+using JetBrains.Annotations;
+
+namespace MetadataExtractor.IO
+{
+    /// <author>Kevin Mott https://github.com/kwhopper</author>
+    /// <author>Drew Noakes https://drewnoakes.com</author>
+    public class ReaderInfo
```

Comment:
Can we give this a better name than `ReaderInfo`. To me, the current name implies it's some immutable bit of data rather than a stateful actionable object.

Should this type be `internal`?

Reply:
It could potentially be internal if the bits we need from it (like global position) are recorded with metadata items. Then the instances can disappear in GC.

This was one of the early class names and I never changed it. Maybe something like "RegionReader" or "DataRegion", etc. I'm open to whatever.

Reply:
I'm also coming up short for name suggestions. Will have a think. If it's internal then it's easier to change later as no one will have taken a dependency on the name.